spot_ros2_control/hardware/include/spot_ros2_control/spot_joint_map.hpp

```diff
@@ -50,64 +61,64 @@ static const std::unordered_map<std::string, size_t> kJointNameToIndexWithoutArm
+/// @return Unordered map that takes joint name to joint index.
+std::unordered_map<std::string, size_t> get_namespaced_joint_map(const std::string& spot_name, bool has_arm) {
+  const auto default_map = has_arm ? kJointNameToIndexWithArm : kJointNameToIndexWithoutArm;
+  if (spot_name.empty()) {
+    return default_map;
+  }
+  std::unordered_map<std::string, size_t> namespaced_map;
+  const std::string joint_prefix = spot_name + "/";
+  for (const auto& pair : default_map) {
+    namespaced_map[joint_prefix + pair.first] = pair.second;
+  }
+  return namespaced_map;
+}
+
 /// @brief Given a list of joints from a JointStates message, put them in the correct order that the Spot Hardware
 /// interface expects.
-/// @param msg JointStates message
-/// @param ordered_joint_angles_ Joint positions from the joint state message following the correct order.
+/// @param input_joint_states The JointStates message received from the robot
+/// @param output_joint_states A JointStates message that will be ordered properly
+/// @param spot_name Namespace that the ros2 control stack was launched in that prefixes the joint names
 /// @return boolean indicating if the joint angles got ordered successfully.
-bool order_joints(const sensor_msgs::msg::JointState& msg, std::vector<double>& ordered_joint_angles) {
-  // this needs to be fixed so that it works if the joints are namespaced
-  const auto njoints = msg.position.size();
-  ordered_joint_angles.resize(njoints);
-  // Different joint index maps for arm-full and arm-less
-  switch (njoints) {
-    // case without arm
-    case kNjointsNoArm:
-      for (size_t i = 0; i < njoints; ++i) {
-        // get the joint name
-        const auto& joint_name = msg.name.at(i);
-        try {
-          const auto joint_index = kJointNameToIndexWithoutArm.at(joint_name);
-          ordered_joint_angles.at(joint_index) = msg.position.at(i);
-        } catch (const std::out_of_range& e) {
-          RCLCPP_INFO_STREAM(rclcpp::get_logger("SpotJointMap"), "Invalid joint: " << joint_name);
-          return false;
-        }
-      }
-      return true;
-
-    // case with arm
-    case kNjointsArm:
-      for (size_t i = 0; i < njoints; i++) {
-        // get the joint name
-        const auto joint_name = msg.name.at(i);
-        try {
-          const auto joint_index = kJointNameToIndexWithArm.at(joint_name);
-          ordered_joint_angles.at(joint_index) = msg.position.at(i);
-        } catch (const std::out_of_range& e) {
-          RCLCPP_INFO_STREAM(rclcpp::get_logger("SpotHardware"), "Invalid joint: " << joint_name);
-          return false;
-        }
-      }
-      return true;
-
-    default:
-      RCLCPP_INFO_STREAM(rclcpp::get_logger("SpotHardware"), "Invalid number of joints: " << njoints);
+bool order_joint_states(const sensor_msgs::msg::JointState& input_joint_states,
+                        sensor_msgs::msg::JointState& output_joint_states, const std::string& spot_name) {
```

Comment:
@khughes-bdai meta: any reason not to have `spot_name`  be the first argument?  Output arguments not showing up last in the signature seems unusual.

Reply:
Good point, updated this in [f01da52](https://github.com/bdaiinstitute/spot_ros2/commit/f01da5284a65ed9c5f465aab18ba6272ae513c50)